test/compilable/noreturn3.d

```diff
@@ -0,0 +1,22 @@
+/*
+REQUIRED_ARGS: -w -o-
```

Comment:
Moved into a separate file because this code triggers an assertion in `cod1.d`.

https://issues.dlang.org/show_bug.cgi?id=21950

Reply:
inspired by 21950 I have searched and found a full hand of other noreturn issues. You might be interested to check if some of them are also fixed by your PR ([21951](https://issues.dlang.org/show_bug.cgi?id=21951), [21952](https://issues.dlang.org/show_bug.cgi?id=21952), [21953](https://issues.dlang.org/show_bug.cgi?id=21953), [21954](https://issues.dlang.org/show_bug.cgi?id=21954), [21955](https://issues.dlang.org/show_bug.cgi?id=21955), [21956](https://issues.dlang.org/show_bug.cgi?id=21956), [21957](https://issues.dlang.org/show_bug.cgi?id=21957),).

Reply:
The second commit will solve the crashes for variables requiring a default initializers (21951).
Also had a look of the other ones, most of them appear to be valid code that isn't supported by the backend (unless my understanding of the DIP is wrong)

Reply:
Thanks for checking them all. 
I'd  suggest to add the test cases for the those who are fixed by this PR and edit the commit msg so that they get closed.

Reply:
and the others require specific PR to lower the expressions to `assert(0);` if I understand correctly.

Reply:
ok

Reply:
#12586